### DISCUSSION TITLE
Feedback on placement of Raven Chat within Desk

### DISCUSSION THREAD

**Discussion by nikkothari22** [The-Commit-Company/raven#607] [Ideas]
**Context**

We are building a small chat interface of Raven within Desk itself so that users can chat quickly without losing context of what they are working on.

It will have the basic features you need for quick texting:
1. Sending/viewing messages with reactions (duh)
2. Sending files
3. Sending current document (you could be on an expense claim for example - and you want to send a message linking that expense claim doc to it so that users know which doc you're referring to)
4. Reacting and replying to messages
5. Notification sound when you receive a message

Features like creating channels, searching for messages, managing channel membership and other advanced bells and whistles will only be available on the web/mobile app. 

Here's a quick POC:

https://github.com/The-Commit-Company/Raven/assets/19825455/c28e8cff-ed4b-46c1-8a6a-0dd8cae2f570

Now, the question is - where do place the chat interface?

We first thought to place it on the right side - like LinkedIn/Facebook do. But, in Frappe, if you collapse a call/email, it also sits in the bottom right. Hence, for now, we're placing the chat interface on the left. Any ideas and suggestions?

**Comment by yahyaoudra** [comment]
This is an amazing feature, I think making a floating bubble could be better.
as v1 just showing you alters with redirection to raven installation link will be better

example 
![raven desk view  V0](https://github.com/The-Commit-Company/Raven/assets/67019341/1477918d-e156-49fb-b09f-450ad4a78ecd)

**Reply by nikkothari22** [reply]
@yahyaoudra We will most likely make this user configurable since some people do not like chat bubbles. Will have to keep it on the left though because of other moving Frappe elements.

**Reply by yahyaoudra** [reply]
@nikkothari22 perfect

**Comment by BenjaminDSmithy** [comment]
Take a page from

[Lark Suite's Messenger](https://www.larksuite.com/en_us/product/messenger)

[Lark Base's Messenger Group Field](https://www.larksuite.com/hc/en-US/articles/428839541567-use-group-fields-in-base)

[Lark Base's Messenger Person Field](https://www.larksuite.com/hc/en-US/articles/360048488182-use-person-fields-in-base)

NOTE: when you select open on either a Person or Group field it directly opens as a slide-out divided window the chat which can also be undocked and floating similar to how Raven operates now with the rollup window
